### PR TITLE
Add system profile for a rawbutton on-off switch

### PR DESCRIPTION
**Pull request by mhilbush**
As discussed with @cweitkamp during my binding review [here](https://github.com/openhab/openhab2-addons/pull/5857#discussion_r331672597), adding a new system profile for a switch whose state changes base on the PRESSED (ON) and RELEASED (OFF) trigger events.

@cweitkamp Is this what you had in mind? 

Signed-off-by: Mark Hilbush <mark@hilbush.com>

### CODE REVIEW

File: bundles/org.openhab.core.thing/src/test/java/org/eclipse/smarthome/core/thing/internal/profiles/RawButtonOnOffSwitchProfileTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.eclipse.smarthome.core.thing.profiles.TriggerProfile;
+import org.eclipse.smarthome.core.types.Command;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+/**
+ *
+ * @author Mark Hilbush - Initial contribution
+ *
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/test/java/org/eclipse/smarthome/core/thing/internal/profiles/RawButtonOnOffSwitchProfileTest.java]: It is just nit-picking: Can you please remove this empty line?

File: bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/SystemProfiles.java [hunk 2]
```diff
@@ -50,6 +51,11 @@
             .withSupportedItemTypes(CoreItemFactory.NUMBER).withSupportedItemTypesOfChannel(CoreItemFactory.NUMBER)
             .build();
 
+    TriggerProfileType RAWBUTTON_ON_OFF_SWITCH_TYPE = ProfileTypeBuilder
+            .newTrigger(RAWBUTTON_ON_OFF_SWITCH, "Raw Button On Off Switch")
+            .withSupportedItemTypes(CoreItemFactory.SWITCH)
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/SystemProfiles.java]: Should we add support for `Dimmer` and `Color` items too as they both accept `OnOffType`.
```suggestion
            .withSupportedItemTypes(CoreItemFactory.SWITCH, CoreItemFactory.DIMMER, CoreItemFactory.COLOR)
```

File: bundles/org.openhab.core.thing/src/test/java/org/eclipse/smarthome/core/thing/internal/profiles/RawButtonOnOffSwitchProfileTest.java (hunk 1 above)
Comment (mhilbush) [reply bundles/org.openhab.core.thing/src/test/java/org/eclipse/smarthome/core/thing/internal/profiles/RawButtonOnOffSwitchProfileTest.java]: Sure. I noticed there's one near the end of one of the other files too. I'll remove that too.

File: bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/SystemProfiles.java (hunk 2 above)
Comment (mhilbush) [reply bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/SystemProfiles.java]: I wasn't sure about this. But, I think you're right that they should be added.

**Comment by mhilbush** [comment]
I updated SystemProfileFactory.java, which I had missed completely... Also, in the test, it looked like it was just the number of profiles that needed to be bumped from 14 to 15.

File: bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/SystemProfileFactory.java [hunk 3]
```diff
@@ -65,17 +65,17 @@
...
+            RAWROCKER_UP_DOWN_TYPE, TIMESTAMP_CHANGE_TYPE, TIMESTAMP_UPDATE_TYPE).collect(Collectors.toSet());
 
     private static final Set<ProfileTypeUID> SUPPORTED_PROFILE_TYPE_UIDS = Stream.of(DEFAULT, FOLLOW, OFFSET,
-            RAWBUTTON_TOGGLE_PLAYER, RAWBUTTON_TOGGLE_PLAYER, RAWBUTTON_TOGGLE_SWITCH, RAWROCKER_DIMMER,
-            RAWROCKER_NEXT_PREVIOUS, RAWROCKER_ON_OFF, RAWROCKER_PLAY_PAUSE, RAWROCKER_REWIND_FASTFORWARD,
-            RAWROCKER_STOP_MOVE, RAWROCKER_UP_DOWN, TIMESTAMP_CHANGE, TIMESTAMP_UPDATE).collect(Collectors.toSet());
+            RAWBUTTON_ON_OFF_SWITCH, RAWBUTTON_TOGGLE_PLAYER, RAWBUTTON_TOGGLE_PLAYER, RAWBUTTON_TOGGLE_SWITCH,
+            RAWROCKER_DIMMER, RAWROCKER_NEXT_PREVIOUS, RAWROCKER_ON_OFF, RAWROCKER_PLAY_PAUSE,
+            RAWROCKER_REWIND_FASTFORWARD, RAWROCKER_STOP_MOVE, RAWROCKER_UP_DOWN, TIMESTAMP_CHANGE, TIMESTAMP_UPDATE)
+            .collect(Collectors.toSet());
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/SystemProfileFactory.java]: One more line in method `createProfile()`. After that we are ready to merge.

Comment (mhilbush) [reply bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/SystemProfileFactory.java]: Ugh! Reading skills are so important. 😦 Should be fixed now.

Comment (cweitkamp) [reply bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/SystemProfileFactory.java]: I submitted #1132 to verify such trouble in the future. :wink:

**Comment by mhilbush** [comment]
> May I ask you to add it to the docs too?

Certainly.

Thanks for your patience in helping me work through this.

**Pull request merged**